### PR TITLE
disable tcmalloc by default when enable jemalloc

### PR DESCRIPTION
**Pull request by xinxinsh**
Signed-off-by: xinxin shu <xinxin.shu@intel.com>

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -420,6 +403,26 @@ AS_IF([test "x$with_jemalloc" = xyes],
...
+	       AC_DEFINE([HAVE_LIBTCMALLOC], [1],
+	       		 [Define if you have tcmalloc])
+	       HAVE_LIBTCMALLOC=1
+	     ],
+	    [AC_MSG_FAILURE(
+		  [no tcmalloc found (use --without-tcmalloc to disable)])])])
+AM_CONDITIONAL(WITH_TCMALLOC, [test "$HAVE_LIBTCMALLOC" = "1"])
+
 # error out if --with-jemalloc and ! --without-tcmalloc
 if test "x$with_jemalloc" = "xyes"; then
```
Comment (shishirng) [inline configure.ac]: This situation is already handled in the following check
# error out if --with-jemalloc and ! --without-tcmalloc
 if test "x$with_jemalloc" = "xyes"; then
         if test "x$with_tcmalloc" != "xno"; then
                   AC_MSG_FAILURE([--with-jemalloc called without --without-tcmalloc])
 	fi
 fi

Comment (xinxinsh) [inline configure.ac]: currently if you enable jemalloc , we need "--with-jemalloc --without-tcmalloc", but with this patch , we just need "--with-jemalloc"

Comment (shishirng) [inline configure.ac]: Sage wanted the build to error out when --with-jemalloc was specified, but not --without-tcmalloc. Hence, the need for both options.

**Comment by liewegas** [comment]
If I remember correct the original concerns were:

- --with-tcmalloc --with-jemalloc would get jemalloc instead of an error
- --with-librocksdb-static --with-jemalloc would get jemalloc but no rocksdb

As long as the result is unambiguous and unsurprising, I think it's fine.  This approach looks ok to me?

**Pull request merged**